02_knight_and_princess/danil_dyachenko/graph_gen.cpp

```diff
@@ -0,0 +1,126 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using VertexId = int;
+
+struct Vertex {
+  VertexId id;
+  std::vector<int> edges;
+};
+
+struct Edge {
+  VertexId id;
+  std::array<int, 2> vertices;
+};
+
+class Graph {
+ public:
+  void add_vertices(const std::vector<Vertex>& vec) { vertices_ = vec; }
```

Comment:
Больше похоже на метод конструктора. Зачем тебе отдельный сеттер для этих данных?
```suggestion
  Graph(const std::vector<Vertex>& vertices) : vertices_(vertices) {}
```